### PR TITLE
separate out __init__ file for metadata pkg

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Some restructuring/cleanup to make #21465 a bit easier

### CODE REVIEW

**Comment by benpankow** [comment]
* **#21468** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21468?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21467** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21467?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21465** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21465?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21576** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21576?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @benpankow and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/dagster/dagster/_core/definitions/metadata/types.py [hunk 1]
```diff
@@ -0,0 +1,957 @@
+import os
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/definitions/metadata/types.py:1]: could this be named `metadata_value.py` for more specificity?

**Review by sryza — DISMISSED** [review]
Praise be! Just one comment.

File: python_modules/dagster/dagster/_core/definitions/metadata/types.py (hunk 1 above)
Comment (benpankow) [reply python_modules/dagster/dagster/_core/definitions/metadata/types.py]: yup

**Review by sryza — APPROVED** [review]

**Pull request merged**